### PR TITLE
v2:interceptors/logging: allow to separate request response payload logging

### PR DESCRIPTION
**Pull request by michaljemala**
Logging payloads of requests on the server-side and similarly of responses on the client-side is a pretty common and desired use-case. However, logging by default payloads of both requests as well as responses in both scenarios is IMO not a very common requirement.

Consider the following PR to enable more granular control of what to log and when. This is achieved by using a special `PayloadDecison` type having the following values:
* `NoPayloadLogging` - Payload logging is disabled.
* `LogRequest` - Only logging of requests is enabled.
* `LogResponse` - Only logging of responses is enabled.
* `LogRequestAndResponse` - Logging of both requests and responses is enabled.
The `PayloadDecison` is to be used instead of the original `bool` by the `logging.ServerPayloadLoggingDecider` and `logging.ClientPayloadLoggingDecider`.

NOTE: I quickly put this together to see whether it could be considered a valid approach, thus please excuse the lack of additional unit tests. I will be glad to add those once the approach is approved.

### CODE REVIEW

**Comment by michaljemala** [comment]
@googlebot I fixed it.

**Comment by michaljemala** [comment]
@googlebot I fixed it.

**Review by johanbrandhorst — COMMENTED** [review]
I quite like this, but it was @bwplotka who refactored these, so I would prefer to have his input.

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:32]: Can we use `switch case` in case we want to add more conditionals later on for different `PayloadDecision`? Would be easy to read and maintain the code 😄

**Review by yashrsharma44 — CHANGES_REQUESTED** [review]
Functionally - I think using a switch case would make this code easier to read and maintainable. Otherwise looks good to me!

File: interceptors/logging/payload.go
Comment (michaljemala) [reply interceptors/logging/payload.go]: Yes sure, good idea. I just changed it.

**Comment by yashrsharma44** [comment]
Hi @michaljemala !

This PR looks good to me, can you run `make lint` for fixing the CI changes required?
Otherwise looks good to me.

**Comment by yashrsharma44** [comment]
cc'ing @bwplotka for the inputs 😄

**Comment by yashrsharma44** [comment]
Seems like the CI build needs approval for first time contributors - https://github.blog/changelog/2021-04-22-github-actions-maintainers-must-approve-first-time-contributor-workflow-runs/

**Comment by michaljemala** [comment]
Hi @yashrsharma44 is there anything else on my side to be done?

**Review by yashrsharma44 — APPROVED** [review]

File: interceptors/logging/logging.go
Comment (bwplotka) [inline interceptors/logging/logging.go:77]:
```suggestion
	LogPayloadRequest
```

File: interceptors/logging/logging.go
Comment (bwplotka) [inline interceptors/logging/logging.go:79]:
```suggestion
	LogPayloadResponse
```

File: interceptors/logging/logging.go
Comment (bwplotka) [inline interceptors/logging/logging.go:81]:
```suggestion
	LogPayloadRequestAndResponse
```

**Review by bwplotka — APPROVED** [review]
LGTM, only minor nit and lint error 🤗 

Thanks!

File: interceptors/logging/logging.go
Comment (michaljemala) [reply interceptors/logging/logging.go]: Done.

File: interceptors/logging/logging.go
Comment (michaljemala) [reply interceptors/logging/logging.go]: Done.

File: interceptors/logging/logging.go
Comment (michaljemala) [reply interceptors/logging/logging.go]: Done.

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**
